cli/lsp/completions.rs

```diff
@@ -452,6 +471,134 @@ fn get_relative_specifiers(
     .collect()
 }
 
+/// Get completions for `npm:` specifiers.
+async fn get_npm_completions(
+  specifier: &str,
+  range: &lsp::Range,
+  npm_registry_url: &Url,
+  file_fetcher: &FileFetcher,
+) -> Option<Vec<lsp::CompletionItem>> {
+  debug_assert!(specifier.starts_with("npm:"));
+  let bare_specifier = &specifier[4..];
+  let get_endpoint = |suffix: &str| {
+    let mut url = npm_registry_url.clone();
+    url
+      .path_segments_mut()
+      .ok()?
+      .pop_if_empty()
+      .extend(suffix.split('/'));
+    Some(url)
+  };
+
+  // First try to match `npm:some-package@<version-to-complete>`.
+  let v_index = bare_specifier.rfind('@');
```

Comment:
> Perhaps instead of this we should use `NpmPackageReqReference` to try to parse the npm specifier first?

Not sure if we can use that since it validates for a completed specifier. But I made the parsing better.